### PR TITLE
Adding option to delay in running the command between sites.

### PR DESCRIPTION
**Pull request by joshirohit100**
cc @vbouchet31 @sdelbosc

### CODE REVIEW

**Comment by sdelbosc** [comment]
What is the purpose of this?

**Comment by joshirohit100** [comment]
@sdelbosc http://jira.alshaya.com:8080/browse/CORE-6611

**Comment by vbouchet31** [comment]
The goal is to have a delay between each sites so if we run drush acsf-tools-ml cr --delay=600, then there will be a 10 minutes delay between each sites. I took cr as an example because it is often the kind of think we don't want to do on all the sites at the same time.

**Comment by vbouchet31** [comment]
@sdelbosc This looks ok to me.

**Comment by sdelbosc** [comment]
But acsf tools does not run anything in parallel (at least for now) so I don't get how we could end up with multiple crons

**Comment by vbouchet31** [comment]
We won't get multiple crons, it will still be procedural with some sleep() between each sites.

**Comment by sdelbosc** [comment]
Seen with Vincent, he aims for a way of running a command on all the sites of the factory but with a pause between two sites to reduce the load, i.e. drush cr, drush search-api-index

Reviewing the code

File: AcsfToolsCommands.php [hunk 1]
```diff
@@ -192,6 +196,12 @@ public function ml($cmd, $command_args = '', $command_options = '', $options = [
 
         $this->output()->writeln("\n=> Running command on $domain");
         drush_invoke_process('@self', $cmd, $command_args, $drush_command_options);
+
```
Comment (sdelbosc) [inline AcsfToolsCommands.php]:
``````
      $i = 0;
      $delay = abs($options['delay']));
      foreach ($sites as $details) {
...
        drush_invoke_process('@self', $cmd, $command_args, $drush_command_options);

        // Delay in running the command for next site.
        if ($delay > 0 && $i < (count($sites) - 1)) {
          $this->output()->writeln("\n=> Sleeping for $delay seconds before running command on next site.");
          sleep($delay);
        }
      }

Comment (nikunjkotecha) [reply AcsfToolsCommands.php]: ❓ why do we need abs() here? also shouldn't we have isset check as it is optional param?

suggestion: we can avoid count($sites) call every-time by doing it outside

I know this is pseudo code but $i is not updated anywhere, can we use index of $sites array?

`foreach ($sites as $i => $details) {`

File: AcsfToolsCommands.php [hunk 2]
```diff
@@ -175,6 +175,8 @@ public function ml($cmd, $command_args = '', $command_options = '', $options = [
         unset($options['profiles']);
       }
 
+      $i = 0;
+      $delay = abs($options['delay']);
```
Comment (sdelbosc) [inline AcsfToolsCommands.php]: I agree with Nikunj on the abs thing. Really needed?
Also make sure we don't have a notice if param is not sent. I guess no as there is a default value.

Comment (joshirohit100) [reply AcsfToolsCommands.php]: `abs()` was here to handle the negative values as `sleep()` works only for positive. As now doing `> 0` test, no need it.

PR updated.

**Pull request merged**